DIPs/DIPNNNN.md

```diff
@@ -0,0 +1,349 @@
+   few most widely used one, making most of its impact transitively.
+5. Having simple and reliable way to add new symbols to runtime will reduce
+   amount of time spent in naming debates and improve readability of resulting
+   APIs.
+6. Attempt to introduce proposed functionality is not a one way ticket because
+   it is intended to not be exposed for public usage initially and thus can
+   be judged (and possibly reverted) again when the time comes.
+
+As an extra benefit, such facility can provide a deprecation path for
+introducing new language keywords if the need arises, though this use case is
+beyond the scope of this DIP and won't be evaluated in details.
+
+### Comparison against other languages
+
+Existing programming languages can be separated in 3 main groups in regards
+of this issue:
+
+#### With single global namespace
+
+Examples: C, early PHP
+
+Such languages tend to rely on [using obscure names](#using-more-obscure-name)
+that makes symbol clash less likely but in general problem is still present.
+When the problem manifests, situation becomes much worse than in D, because such
+languages lack tools to disambiguate name clash explicitly.
+
+#### With namespaces, allowing non-qualified access
+
+Examples: C++, C#
+
+This category is most simple to D, meaning that language provides tools to
+disambiguate any names but at the same time allows imports with non-qualified
+access to imported symbols.
+
+In D this is default `import` behavior.
+
+In C++ : `using namespace Namespace.Subnamespace`.
+
+In C# : `using Namespace.Subnamespace`
+
+Importance of problem varies depending on idiomatic code style dominant for the
+language and other available tools. For example, C# documentation
+[recommends](https://msdn.microsoft.com/en-us/library/dfb3cx8s.aspx) using
+dedicated namespace access operator `::` instead of uniform `.` to reduce the
+risk of clashes.
+
+However, there are few D specifics that make issues somewhat more important:
+
+- In all mentioned languages but D it is common convention to
+  only use unqualified access for symbols in standard library. In D plain
+  `import` is most widespread way to use all libraries, including dub registry.
+- D runtime features module `object.d` which is implicitly imported everywhere,
+  greatly increasing chance for name clash for any symbol in it.
+
+#### With namespaces, restricted non-qualified access
+
+Examples: Rust, Go
+
+In both Rust and Go normal imports still require using module name for
+qualification. For example, in Rust `use Namespace::Subnamespace::Module` still
+requires to explicitly write `Module::foo()` to call `foo` function.
+
+Both languages provide a way to enable unqualified access (`import . "Module"`
+in Go, `use Namespace::Module::*` in Rust) but this feature is heavily
+discouraged from usage in production projects exactly because of problems
+described in this DIP.
+
+This completely solves the issue described in this DIP but is too different
```

Comment:
s/This completely/Using qualified lookups consistently/
s/is too different/such an approach is too different/